src/AlgebraicGeometry/ToricVarieties/ToricSchemes/attributes.jl

```diff
@@ -0,0 +1,136 @@
+#######################################
+### 1: Underlying scheme
+#######################################
+
+@doc raw"""
+    underlying_scheme(X::AffineNormalToricVariety)
+
+For an affine toric scheme ``X``, this returns
+the underlying scheme. In other words, by applying
+this method, you obtain a scheme that has forgotten
+its toric origin.
+
+# Examples
+```jldoctest
+julia> C = positive_hull([1 0; 0 1])
+Polyhedral cone in ambient dimension 2
+
+julia> antv = affine_normal_toric_variety(C)
+Normal, affine toric variety
+
+julia> underlying_scheme(antv)
+Spectrum
+  of quotient
+    of multivariate polynomial ring in 2 variables over QQ
+    by ideal(0)
+```
+"""
+@attr Spec{QQField, MPolyQuoRing{QQMPolyRingElem}} underlying_scheme(X::AffineNormalToricVariety) = Spec(base_ring(toric_ideal(X)), toric_ideal(X))
+
+
+@doc raw"""
+    underlying_scheme(X::NormalToricVariety)
+
+For a toric covered scheme ``X``, this returns
+the underlying scheme. In other words, by applying
+this method, you obtain a scheme that has forgotten
+its toric origin.
+
+# Examples
+```jldoctest
+julia> P2 = projective_space(NormalToricVariety, 2)
+Normal, non-affine, smooth, projective, gorenstein, fano, 2-dimensional toric variety without torusfactor
+
+julia> underlying_scheme(P2)
+Scheme
+  over rational field
+with default covering
+  described by patches
+    1: normal, affine toric variety
+    2: normal, affine toric variety
+    3: normal, affine toric variety
+  in the coordinate(s)
+    1: [x_1_1, x_2_1]
+    2: [x_1_2, x_2_2]
+    3: [x_1_3, x_2_3]
+```
+"""
+@attr function underlying_scheme(Z::NormalToricVariety)
+  patch_list = affine_open_covering(Z)
+  for (k, A) in enumerate(patch_list)
+    C = cone(pm_object(A).WEIGHT_CONE)
+    n = length(hilbert_basis(C))
+    R, _ = polynomial_ring(QQ, ["x_$(i)_$(k)" for i in 1:n], cached = false);
+    set_attribute!(A, :toric_ideal, toric_ideal(R, A))
+  end
+  cov = Covering(patch_list)
+  
+  for i in 1:(length(patch_list)-1)
+    for j in i+1:length(patch_list)
+      
+      X = patch_list[i]
+      Y = patch_list[j]
+      
+      # Step 1: Only glue if the cone of intersection has the "right" dimension:
+      facet = intersect(cone(X), cone(Y))
+      (dim(facet) == dim(cone(X)) - 1) || continue
```

Comment:
Then you need to assume pureness.

Furthermore this intersection is expensive and it can be read off easily from the Hasse diagram.

Reply:
Thank you for spotting this. Shall we then demand `pureness`  of the fan to begin with and error if violated (like "underlying_scheme is currently only supported for toric varieties with pure fans")?

I recall that we had a discussion of the Hasse diagrams almost one year back. How can we read this off from the Hasse diagram (as in, what code do we have to execute for this)?

Reply:
I have just pushed https://github.com/oscar-system/Oscar.jl/pull/2700/commits/7930b7c72a7c57bfa7a8b75a375cae9c8ee03778, which requires the fan to be pure for the construction of `underlying_scheme` of a normal toric variety.

As discussed on slack, I will open an issue now, so we do not forget to use the Hasse diagram as you are suggesting.

Reply:
Issue is here: https://github.com/oscar-system/Oscar.jl/issues/2718